cli/command/swarm/ca.go

```diff
@@ -61,6 +61,10 @@ func runRotateCA(dockerCli command.Cli, flags *pflag.FlagSet, opts caOptions) er
 	}
 
 	if !opts.rotate {
+		if flags.Changed(flagCACert) || flags.Changed(flagCAKey) || flags.Changed(flagCertExpiry) {
+			fmt.Fprintln(dockerCli.Out(), "Warning: because the `--rotate` flag was not provided, all other flags will be ignored")
+		}
```

Comment:
@dnephin It just didn't particularly seem like an error to me, although I don't feel very strongly.  

> I notice `flagExternalCA` is not being checked. Is that right?

Ah sorry, that is also being addressed by `mergeSwarmSpec`.  I'll add that to the comment.